src/Xamarin.Android.Sdk/Sdk/Sdk.targets

```diff
@@ -0,0 +1,3 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildThisFileDirectory)..\targets\Xamarin.Android.Sdk.targets"  />  
```

Comment:
Where does this `targets` directory/convention come from?  Is it an artifact of SDK Short-form projects?

Reply:
This was done to follow `Microsoft.Build.Sdk` behavior:
https://github.com/dotnet/sdk/tree/master/src/Tasks/Microsoft.NET.Build.Tasks/targets

Reply:
Yeah, but....

This is where my brain starts exploding.

As far as I can tell, `Microsoft.NET.Build.Tasks/targets` works because they use a wildcard within their `.csproj`:

https://github.com/dotnet/sdk/blob/c60cea4d72e90efac857a86a2215a590e77964b0/src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj#L77

```xml
  <ItemGroup>
    <None Include="targets\**\*" PackagePath="targets\" />
  </ItemGroup>
```

Fortunately (?), this did lead me to the `%(PackagePath)` metadata -- which is presumably documented somewhere?  are there any good NuGet packaging docs? -- which is used in `Packager.csproj`:

https://github.com/pjcollins/xamarin-android/blob/c5e291f709cd41901d58f1a1dc00f5f53059fec2/build-tools/create-packs/Packager.csproj#L55-L69

...though this still leaves me partially confused (need more docs & context!): is `targets` a "conventional" directory in NuGet packages?  (Clearly I haven't unzipped enough of them.)  As the above fragment *appears* to place `src\Xamarin.Android.Sdk\targets\**\*` into a `PackagePath` of `/targets/**/*`...

Reply:
For conventional NuGet packages, I think the `build` folder would be the expected/preferred location for targets/props folders and any custom build tasks, as is documented here:

https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#from-a-convention-based-working-directory

However, this package represents an MSBuild Project SDK, the directory hierarchy of which doesn't have any good documentation that I could find. I've been _roughly_ following how the `Microsoft.NET.Sdk` is laid out, for instance on macOS:

```
/usr/local/share/dotnet/sdk/3.1.100/Sdks/Microsoft.NET.Sdk/tools/netcoreapp2.1/*
/usr/local/share/dotnet/sdk/3.1.100/Sdks/Microsoft.NET.Sdk/tools/net472/*
/usr/local/share/dotnet/sdk/3.1.100/Sdks/Microsoft.NET.Sdk/targets/*
/usr/local/share/dotnet/sdk/3.1.100/Sdks/Microsoft.NET.Sdk/Sdk/Sdk.targets
/usr/local/share/dotnet/sdk/3.1.100/Sdks/Microsoft.NET.Sdk/Sdk/Sdk.props
...
````

The `targets/*` path above "works" because the files in here are explicitly imported through the main `Sdk/Sdk.props`  and `Sdk/Sdk.targets` entry points: 
https://github.com/dotnet/sdk/blob/c60cea4d72e90efac857a86a2215a590e77964b0/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.props#L41
https://github.com/dotnet/sdk/blob/c60cea4d72e90efac857a86a2215a590e77964b0/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.targets#L19

Below is the current structure of the Xamarin.Android.Sdk.nupkg which contains both macOS and Windows components. The folder hierarchy is a bit arbitrary at the moment, because we will be looking to break this down further into an SDK (the current `Sdk`, `targets` and `tools` folders), and targeting/framework/runtime packs (the current `lib` folder) in the ~near future:
 
https://gist.github.com/pjcollins/e3ce7616bcea6d343aff6d607e07d4f7

Reply:
There are a few ways you can import an SDK

global.json
```json
{
    "msbuild-sdks":
    {
         "Xamarin.Sdk": "1.0.0"
    }
}
```
or installed through `$(MSBuildExtensionsPath)/Sdks` or use your own `MSBuildSdksPath` env var, which will lose access to the former path. If you don't provide the version, the resolver (`MSBuildSdkResolver`) will search through the Sdks paths defined earlier.
```xml
<Project Sdk="Xamarin.Sdk" />

<Sdk Name="Xamarin.Sdk" />

<Import Sdk="Xamarin.Sdk" Project="Sdk.props">
<Import Sdk="Xamarin.Sdk" Project="Sdk.targets">
<Import Sdk="Xamarin.Sdk" Project="Sdk.Custom.file">
<Import Sdk="Xamarin.Sdk" Project="..\targets\Xamarin.Sdk.props">
```
--- OR ---

providing explicit version: (will search through NuGet through `NuGetSdkResolver`)
```xml
<Project Sdk="Xamarin.Sdk/1.0.0" />
<Sdk Name="Xamarin.Sdk/1.0.0">
<Import Sdk="Xamarin.Sdk/1.0.0" Project="...">
```

pinging @rainersigwald to verify...

Reply:
@Nirmal4G we’ve progressed quite beyond this PR at this point, thanks!

You might learn more about what we are doing in this doc, we are adding to it as we go: https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/DotNet5.md

If you review our current PRs with the title [.NET 5] those would also be of interest.

Reply:
Sure, Will do!